inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -1319,51 +1320,106 @@ struct jit_uni_reduce_post_kernel_f32 : public jit_uni_reduce_post_kernel, publi
     }
 };
 
-MKLDNNReduceNode::MKLDNNReduceNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
-        : MKLDNNNode(layer, eng, cache) {}
+static bool getKeepDims(const std::shared_ptr<ngraph::Node>& op) {
+    if (const auto reduce = std::dynamic_pointer_cast<ngraph::op::util::ArithmeticReductionKeepDims>(op)) {
+        return reduce->get_keep_dims();
+    } else if (const auto reduce = std::dynamic_pointer_cast<ngraph::op::util::LogicalReductionKeepDims>(op)) {
+        return reduce->get_keep_dims();
+    } else {
+        THROW_IE_EXCEPTION << "Reduce node with name " << op->get_friendly_name()
```

Comment:
`is incompatibility with ...` -> `is not derived from ...`

Reply:
fixed